test/FunctionalTests/Microsoft.OData.Core.Tests/Build.NetFramework/ODataMetadataSelectorTests.cs

```diff
@@ -0,0 +1,223 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="JsonFullMetadataLevelTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.OData.Edm;
+using Microsoft.OData.Evaluation;
+using Microsoft.OData.JsonLight;
+using Microsoft.OData.Tests.Evaluation;
+using Microsoft.Test.OData.Utils.Metadata;
+using Xunit;
+using Microsoft.OData.Tests.UriParser;
+
+namespace Microsoft.OData.Tests.JsonLight
+{
+    public class ODataMetadataSelectorTests
+    {
+        private static readonly Uri MetadataDocumentUri;
+        private readonly static JsonFullMetadataLevel fullMetadataLevel;
+
+        private readonly static JsonMinimalMetadataLevel minimalMetadataLevel = new JsonMinimalMetadataLevel();
+
+        private static IEdmModel Model;
+        private static readonly TestFeedAndEntryTypeContext personTypeContext;
+        private static readonly TestFeedAndEntryTypeContext managerTypeContext;
+        private static readonly TestFeedAndEntryTypeContext dogStreamContext;
+        private static ODataResource resource;
+
+        private const string EntitySetName = "People";
+        private const string EntityTypeName = "Fully.Qualified.Namespace.Person";
+        private const string DerivedEntitySetName = "Manager";
+        private const string DerivedEntityTypeName = "Fully.Qualified.Namespace.Manager";
+        private const string DogEntitySetName = "Dog";
+        private const string DogEntityTypeName = "Fully.Qualified.Namespace.Dog";
+
+
+
+        static ODataMetadataSelectorTests()
+        {
+            MetadataDocumentUri = new Uri("http://submodel1/", UriKind.Absolute);
+
+            GenerateModel();
+            
+            resource = new ODataResource();
+            fullMetadataLevel = new JsonFullMetadataLevel(MetadataDocumentUri, Model);
+            minimalMetadataLevel = new JsonMinimalMetadataLevel();
+
+            personTypeContext = new TestFeedAndEntryTypeContext
+            {
+                NavigationSourceName = EntitySetName,
+                NavigationSourceEntityTypeName = EntityTypeName,
+                ExpectedResourceTypeName = EntityTypeName,
+                IsMediaLinkEntry = false,
+                IsFromCollection = false,
+                NavigationSourceKind = EdmNavigationSourceKind.EntitySet
+            };
+
+            managerTypeContext = new TestFeedAndEntryTypeContext
+            {
+                NavigationSourceName = DerivedEntitySetName,

```

Comment:
>NavigationSourceName = DerivedEntitySetName, [](start = 16, length = 44)

Should this really be a different entity set, or should it still be EntitySetName with DerivedEntityTypeName?